### PR TITLE
Update EventSource definitions to be compatible with lib.es6.d.ts

### PR DESCRIPTION
**Pull request by afroozeh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.mozilla.org/en-US/docs/Web/API/EventSource
https://github.com/Microsoft/TypeScript/blob/d579793d0aba0b531fd0edc8fa3e5e3df2bd4140/lib/lib.es6.d.ts#L10465
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by scottleedavis — APPROVED** [review]
Looks great!

File: types/eventsource/index.d.ts [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 // Definitions by: Scott Lee Davis <https://github.com/scottleedavis>
+//                 Ali Afroozeh <https://github.com/afroozeh>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 2.8
 
 declare class EventSource {
   constructor(url: string, eventSourceInitDict?: EventSource.EventSourceInitDict);
-
-  static CLOSED: EventSource.ReadyState;
```
Comment (rbuckton) [inline types/eventsource/index.d.ts]: `CLOSED`, `CONNECTING`, and `OPEN` are available on both the constructor (`static`) and the prototype.

File: types/eventsource/lib/eventsource-polyfill/index.d.ts [hunk 2]
```diff
@@ -1,16 +1,17 @@
 declare class EventSource {
   constructor(url: string, eventSourceInitDict?: EventSource.EventSourceInitDict);
-
-  static CLOSED: EventSource.ReadyState;
```
Comment (rbuckton) [inline types/eventsource/lib/eventsource-polyfill/index.d.ts]: `CLOSED`, `CONNECTING`, and `OPEN` are available on both the constructor (`static`) and the prototype.

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Comment by scottleedavis** [comment]
@rbuckton  does this look ok now?

**Pull request merged**